clever/launch/main_camera.launch

```diff
@@ -1,20 +1,19 @@
 <launch>
-    <!-- Camera position and orientation are represented by base_link -> main_camera_optical transform -->
-    <!-- static_transform_publisher arguments: x y z yaw pitch roll frame_id child_frame_id -->
-
     <!-- article about camera setup: https://clever.coex.tech/camera_frame -->
 
-    <!-- camera is oriented downward, camera cable goes backward [option 1] -->
-    <node pkg="tf2_ros" type="static_transform_publisher" name="main_camera_frame" args="0.05 0 -0.07 -1.5707963 0 3.1415926 base_link main_camera_optical"/>
+    <arg name="orientation_z" default="down"/> <!-- direction the camera points: down, up -->
+    <arg name="orientation_y" default="backward"/> <!-- direction the camera cable points: backward, forward -->
 
-    <!-- camera is oriented downward, camera cable goes forward  [option 2] -->
-    <!--<node pkg="tf2_ros" type="static_transform_publisher" name="main_camera_frame" args="0.05 0 -0.07 1.5707963 0 3.1415926 base_link main_camera_optical"/>-->
+    <!-- Camera position and orientation are represented by base_link -> main_camera_optical transform -->
+    <!-- static_transform_publisher arguments: x y z yaw pitch roll frame_id child_frame_id -->
 
-    <!-- camera is oriented upward, camera cable goes backward   [option 3] -->
-    <!--<node pkg="tf2_ros" type="static_transform_publisher" name="main_camera_frame" args="0.05 0 0.07 1.5707963 0 0 base_link main_camera_optical"/>-->
+    <node if="$(eval orientation_z == 'down' and orientation_y == 'backward')" pkg="tf2_ros" type="static_transform_publisher" name="main_camera_frame" args="0.05 0 -0.07 -1.5707963 0 3.1415926 base_link main_camera_optical"/>
+    <node if="$(eval orientation_z == 'down' and orientation_y == 'forward')" pkg="tf2_ros" type="static_transform_publisher" name="main_camera_frame" args="0.05 0 -0.07 -1.5707963 0 3.1415926 base_link main_camera_optical"/>
+    <node if="$(eval orientation_z == 'up' and orientation_y == 'backward')" pkg="tf2_ros" type="static_transform_publisher" name="main_camera_frame" args="0.05 0 -0.07 -1.5707963 0 3.1415926 base_link main_camera_optical"/>
+    <node if="$(eval orientation_z == 'up' and orientation_y == 'forward')" pkg="tf2_ros" type="static_transform_publisher" name="main_camera_frame" args="0.05 0 -0.07 -1.5707963 0 3.1415926 base_link main_camera_optical"/>
```

Comment:
I believe the shifts should be changed as well. A camera that is pointing upward is also usually shifted by a non-trivial amount, same goes for forward/backward orientation.

Reply:
Now that I look closely, all orientations are actually the same. I believe this was not the intention.

Reply:
Yeah, shift should be corrected, and the orientation values are incorrect as well.

Reply:
Fixed.